RevenueCatUI/CustomerCenter/ViewModels/CustomerCenterViewModel.swift

```diff
@@ -107,4 +117,19 @@ import RevenueCat
         }
     }
 
+    func performRestore() async -> RestorePurchasesAlert.AlertType {
+        self.customerCenterActionHandler?.onRestoreStarted()
+        do {
+            let customerInfo = try await Purchases.shared.restorePurchases()
```

Comment:
In order to be able to mock this easily when we start writing more tests I think we can have a `purchasesProvider` like the one we have in `ManageSubscriptionsViewModel` that has the functions that are being used in this view model. Something like:

```
protocol CustomerCenterPurchasesType: Sendable {

    @Sendable
    func customerInfo() async throws -> CustomerInfo

    @Sendable
    func restorePurchases() async throws -> CustomerInfo

}
```

If we do that, we can remove the `private var customerInfoFetcher: CustomerInfoFetcher` in this class and mock the return of `customerInfo()` using a `MockCustomerCenterPurchasesType`

By the way the `PurchasesType` comes from a `PaywallPurchasesType` in `Paywalls`. That's where I got it from.

Reply:
Talked about this and we are going to unify the different abstractions into a single one for CustomerCenter, since we will need a few, and it was getting complex. However, will do this on a separate PR.